src/resource/ResourceLoader.cpp

```diff
@@ -40,21 +42,53 @@ std::shared_ptr<Window> ResourceLoader::GetContext() {
 }
 
 std::shared_ptr<Resource> ResourceLoader::LoadResource(std::shared_ptr<OtrFile> fileToLoad) {
+    auto stream = std::make_shared<MemoryStream>(fileToLoad->Buffer.get(), fileToLoad->BufferSize);
+    auto reader = std::make_shared<BinaryReader>(stream);
     std::shared_ptr<Resource> result = nullptr;
+    ResourceType resourceType;
+    uint64_t id = 0xDEADBEEFDEADBEEF;
 
-    if (fileToLoad != nullptr) {
-        auto stream = std::make_shared<MemoryStream>(fileToLoad->Buffer.data(), fileToLoad->Buffer.size());
-        auto reader = std::make_shared<BinaryReader>(stream);
+    // Determine if file is binary or XML...
```

Comment:
First byte is endian order. We should be ready this as the second byte to maintain OTR compatibility.